packages/beacon-node/src/chain/historicalState/worker.ts

```diff
@@ -0,0 +1,176 @@
+import worker from "node:worker_threads";
+import {Transfer, expose} from "@chainsafe/threads/worker";
+import {createBeaconConfig, chainConfigFromJson} from "@lodestar/config";
+import {getNodeLogger} from "@lodestar/logger/node";
+import {
+  BeaconStateTransitionMetrics,
+  EpochTransitionStep,
+  PubkeyIndexMap,
+  StateCloneSource,
+  StateHashTreeRootSource,
+} from "@lodestar/state-transition";
+import {LevelDbController} from "@lodestar/db";
+import {RegistryMetricCreator, collectNodeJSMetrics} from "../../metrics/index.js";
+import {JobFnQueue} from "../../util/queue/fnQueue.js";
+import {QueueMetrics} from "../../util/queue/options.js";
+import {BeaconDb} from "../../db/index.js";
+import {HistoricalStateWorkerApi, HistoricalStateWorkerData} from "./types.js";
+import {getHistoricalState} from "./getHistoricalState.js";
+
+// most of this setup copied from networkCoreWorker.ts
+
+const workerData = worker.workerData as HistoricalStateWorkerData;
+
+// TODO: Pass options from main thread for logging
+// TODO: Logging won't be visible in file loggers
+const logger = getNodeLogger(workerData.loggerOpts);
+
+logger.info("Historical state worker started");
+
+const config = createBeaconConfig(chainConfigFromJson(workerData.chainConfigJson), workerData.genesisValidatorsRoot);
+
+const db = new BeaconDb(config, await LevelDbController.create({name: workerData.dbLocation}, {logger}));
+
+const abortController = new AbortController();
```

Comment:
![36vli2-723574080](https://github.com/ChainSafe/lodestar/assets/18608739/a733e34e-2a8d-43ad-8dcf-93457d607943)

Reply:
Why do we actually need this AbortController here if it will not be synchronized with the main thread.  The kill signal comes from the outside

Reply:
I suppose we don't need it exactly, its just convenient as a general pattern to attach cleanup fns to the abort signal and trigger the abort in `close`.